test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -962,14 +967,23 @@ private Set<String> getAllUnexpectedTemplates() throws IOException {
      */
     @SuppressWarnings("unchecked")
     protected void deleteAllNodeShutdownMetadata() throws IOException {
-        if (has(ProductFeature.SHUTDOWN) == false || minimumNodeVersion().before(Version.V_7_15_0)) {
-            // Node shutdown APIs are only present in xpack
+        if (has(ProductFeature.SHUTDOWN) == false) {
             return;
         }
+
         Request getShutdownStatus = new Request("GET", "_nodes/shutdown");
         Map<String, Object> statusResponse = responseAsMap(adminClient().performRequest(getShutdownStatus));
-        List<Map<String, Object>> nodesArray = (List<Map<String, Object>>) statusResponse.get("nodes");
-        List<String> nodeIds = nodesArray.stream().map(nodeShutdownMetadata -> (String) nodeShutdownMetadata.get("node_id")).toList();
+
+        Object nodesResponse = statusResponse.get("nodes");
+        final List<String> nodeIds;
+        if (nodesResponse instanceof List<?>) { // `nodes` is parsed as a List<> only if it's populated (not empty)
+            assert minimumNodeVersion().onOrAfter(Version.V_7_15_0);
```

Comment:
Shouldn't this be a feature check?

Reply:
This assert is here just to double-check the behaviour is not changed. I will remove this assert with the other version usage when we will have feature checks in rest tests.